libraries/apollo-runtime/src/appleMain/kotlin/com/apollographql/apollo3/network/ws/NSURLSessionWebSocketEngine.kt

```diff
@@ -41,10 +49,16 @@ actual class DefaultWebSocketEngine(
 
   actual constructor() : this(
       webSocketFactory = { request, connectionListener ->
+        // We must not use the main thread's queue for the delegateQueue
```

Comment:
That's the fix for the WebSocket not working on Apple (regression introduced in #4287)